libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -356,6 +383,15 @@ bool AP_BattMonitor::overpower_detected(uint8_t instance) const
 #endif
 }
 
+bool AP_BattMonitor::has_cell_voltages(const uint8_t instance) const
+{
+    if (instance < _num_instances && drivers[instance] != nullptr) {
+        return drivers[instance]->has_cell_voltages();
```

Comment:
This is insufficient. It's possible to have a SMBus battery that does't use the fields we are requesting for cell voltages (or doesn't implement them). You need to check that if the driver is capable of fetching a per cell voltage that the cell voltages in the state have actually been set.